src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -679,6 +684,23 @@ public Topic getDeploymentPollingFrequencySeconds() {
         return getTopic(DEPLOYMENT_POLLING_FREQUENCY_SECONDS);
     }
 
+    /**
+     * Get s3 endpoint topic.
+     *
+     * @return s3 endpoint topic
+     */
+    public Topic gets3EndpointType() {
+        Topic s3EndpointType = getTopic(S3_ENDPOINT_TYPE).dflt("");
+        String s3EndpointSysProp = System.getProperty(SdkSystemSetting.AWS_S3_US_EAST_1_REGIONAL_ENDPOINT.property());
+
+        if (Coerce.toString(s3EndpointType).isEmpty()
+                && s3EndpointSysProp.equalsIgnoreCase(S3_ENDPOINT_REGIONAL_TYPE)) {
+            //System property already set to regional when installing Nucleus
+            return s3EndpointType.withValue(S3_ENDPOINT_REGIONAL_TYPE);
+        }
+        return getTopic(S3_ENDPOINT_TYPE).dflt(S3_ENDPOINT_DEFAULT_TYPE);
```

Comment:
This is not going to `dflt` right? because you already set it to empty string. Can you also make sure you test this path?

Reply:
Updated